src/api/firebase.js

```diff
@@ -87,10 +91,50 @@ export async function updateItem(listId, itemId, checked) {
 	const listItemRef = doc(db, listId, itemId);
 	const listItemSnap = await getDoc(listItemRef);
 
-	let totalPurchases = listItemSnap.data().totalPurchases;
+	// Retrieve item property values stored in Firebase
+	let { totalPurchases, dateLastPurchased, dateCreated, dateNextPurchased } =
+		listItemSnap.data();
+
+	//This function returns the next purchase date in milliseconds
+	const getNextPurchaseDate = () => {
+		// Function estimating previous purchase interval
+		const previousEstimate = dateLastPurchased
+			? getDaysBetweenDates(dateNextPurchased, dateLastPurchased)
+			: getDaysBetweenDates(dateNextPurchased, dateCreated);
+
```

Comment:
check the value of `previousEstimate`here; the number you are getting is extremely small. i believe what you wanted to pass was `getDaysBetweenDates(dateNextPurchased.toDate(), dateLastPurchased.toDate())`

Reply:
Made this change! We're still working on figuring out the bugs that made dateNextPurchased the same day as dateLastPurchased, but this fix may have solved it!

Reply:
Okay going back on what I just said! We played around with the date next purchased if the date created and date purchased are the same day, and looking at calculateEstimate, it intentionally does that because it assumes the user buys it often.

Reply:
@sarahmjenkins can you please take another look? i liked your initial check for dateLastPurchased, 
`const previousEstimate = dateLastPurchased
			? getDaysBetweenDates(dateNextPurchased.toDate(), dateLastPurchased.toDate())
			: getDaysBetweenDates(dateNextPurchased.toDate(), dateCreated.toDate());`